program-runtime/src/loaded_programs.rs

```diff
@@ -423,241 +543,267 @@ mod tests {
         )
     }
 
+    fn insert_unloaded_program(
+        cache: &mut LoadedPrograms,
+        key: Pubkey,
+        slot: Slot,
+    ) -> Arc<LoadedProgram> {
+        let unloaded = Arc::new(LoadedProgram {
```

Comment:
How about constructing a `TestLoaded` here and then turning it into an unloaded entry? Because in the production code `.to_unloaded()` is the only way to get such an entry.